### PR TITLE
Add `set_u!` method

### LINKED ISSUE #206: `set_u!` with symbols
Currently modifying the state of an integrator requires passing a `Vector` to `set_u!`, which is nontrivial when dealing with larger systems. Is there a way to do this symbolically, that I can't find? E.g. `set_u!(integrator, x, 1.23)`? Would this be feasible to implement?

### PR DESCRIPTION
**Pull request by AayushSabharwal**
Closes #206

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
PR to the integrator docs as well.

**Comment by AayushSabharwal** [comment]
I've added the test to `downstream/integrator_indexing.jl` for now. Should I let it be there or move it somewhere else?

**Comment by AayushSabharwal** [comment]
Shouldn't rebuilding the integrator docs just pick up the additional docstring for `set_u!` or is there something else I should add as well?

File: src/integrator_interface.jl [hunk 1]
```diff
@@ -332,6 +332,28 @@ function set_u!(integrator::DEIntegrator, u)
     error("set_u!: method has not been implemented for the integrator")
 end
 
+"""
+    set_u!(integrator::DEIntegrator, sym, val)
+
+Set current state of symbol `sym` in `integrator` to `val`.
+"""
```
Comment (ChrisRackauckas) [inline src/integrator_interface.jl:339]: Make a `function set_u! end` and just put the docstrings together.

**Review by ChrisRackauckas — APPROVED** [review]

**Pull request merged**